common.blocks/select/_mode/select_mode_radio.bemhtml

```diff
@@ -1,12 +1,12 @@
 block('select').mod('mode', 'radio')(
-    def().match(this._checkedOptions)(function() {
+    def().match(function() { return this._checkedOptions; })(function() {
         var checkedOptions = this._checkedOptions,
             firstOption = this._firstOption;
         if(firstOption && !checkedOptions.length) {
             firstOption.checked = true;
             checkedOptions = [firstOption];
         }
-        applyNext({ _checkedOption : checkedOptions[0] });
+        return applyNext({ _checkedOption : checkedOptions[0] });
     }),
 
     content()(function() {
```

Comment:
@tadatuta very interesting!

@veged the situation is following: there are `block('select').mod('mode', 'radio')` which has higher priority than `block('select').def()` in the `.bemhtml` file. When `block('select').def()` calls `applyNext()` - it is only invoking the matches in the blocks with lower priority than `block('select').def()`, skipping the `block('select').mod('mode', 'radio')`.

Looks like our idea about incrementing index and invoking next templates is not as **viable** as we thought it will be. What do you think?